azure/scope/machinepool_test.go

```diff
@@ -98,6 +98,58 @@ func TestMachinePoolScope_Name(t *testing.T) {
 		})
 	}
 }
+
+func TestMachinePoolScope_ProviderID(t *testing.T) {
+	tests := []struct {
+		name             string
+		machinePoolScope MachinePoolScope
+		want             string
+	}{
+		{
+			name: "valid providerID",
+			machinePoolScope: MachinePoolScope{
+				AzureMachinePool: &infrav1exp.AzureMachinePool{
+					Spec: infrav1exp.AzureMachinePoolSpec{
+						ProviderID: "azure:///subscriptions/123/resourceGroups/rg/providers/Microsoft.Compute/virtualMachines/vm",
```

Comment:
can we add a unit test with a VMSS provider ID? Would be good to have both coverage for both uniform and flex mode

Reply:
Sure, good idea.

Reply:
Updated with test cases specifically for VMSS Uniform and Flex strings.